### ISSUE TITLE
How to scan more than 3 QR codes at the same time ?

### ISSUE THREAD

**Issue by elapouya** [googlesamples/android-vision#370]
I want to make massive QR code scanning with Google Mobile Vision. I tried your sample APP,  But it can track only 3 qr code at the same time : How can I increase this limit ?

Note : I tried to scan this picture : https://www.flickr.com/photos/periodicvideos/5915143448/in/album-72157627314434233/

**Comment by pchx** [comment]
In order to keep latency down and maintain responsiveness, the Barcode API only processes a limited number of QR finder patterns, which in turn limits the number of QR codes that it will read.

Unfortunately, this limit cannot be changed at runtime.

**Comment by thethoughtpolice** [comment]
Thanks for the information - that helps explain the issue.  Can you suggest anything I can do - for example varying QR pattern/types, sizes....anything?!  I really need 4 QRs (or maybe even 4 barcodes) as I use their positions to further process the document content

**Comment by pchx** [comment]
No. If you are looking for fiducial markers, you might try using [AZTEC codes](https://en.wikipedia.org/wiki/Aztec_Code) (supported by our API) or the [AprilTags library](https://april.eecs.umich.edu/software/apriltag.html).

**Comment by thethoughtpolice** [comment]
Thanks for the advice.   I'll look into the AZTEC codes...do they have the same limitation in terms of number I can read in one pass?

**Comment by pchx** [comment]
No. You shouldn't have any problem reading four AZTEC codes in a single pass.

**Comment by thethoughtpolice** [comment]
4 AZTEC codes look like they'll be fine after some preliminary testing - thanks for your help @pchx

**Comment by thethoughtpolice** [comment]
@pchx - I've looking a bit further into this and part of the overall solutions uses another barcode detection framework which doesn't appear to support AZTEC codes (or AprilTags).  So a bit of a dilemma now as I really need multi-QR approach (or face writing a huge change).  Do you imagine there will be any further work to permit 4 QRs to be collected?  If not, looks like I've (very reluctantly) have to look at third party software.
